### PR TITLE
created a doc page on ml models and model types

### PR DESCRIPTION
**Pull request by martyna-mindsdb**
## Description

- created a doc page on ml models and model types
[ML Models in MindsDB - MindsDB - Case 1.pdf](https://github.com/mindsdb/mindsdb/files/10240057/ML.Models.in.MindsDB.-.MindsDB.-.Case.1.pdf)
[ML Models in MindsDB - MindsDB - Case 2.pdf](https://github.com/mindsdb/mindsdb/files/10240063/ML.Models.in.MindsDB.-.MindsDB.-.Case.2.pdf)
[ML Models in MindsDB - MindsDB - Case 3.pdf](https://github.com/mindsdb/mindsdb/files/10240069/ML.Models.in.MindsDB.-.MindsDB.-.Case.3.pdf)
- updated mint.json with a new navigation item

**Fixes** #4181 

## Type of change

- [ ] ⚡ New feature (non-breaking change which adds functionality)

### What is the solution?

As in Description.

## Checklist:

- [ ] I have updated the documentation.

### CODE REVIEW

File: docs/model-types.mdx [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+SELECT *
+FROM mindsdb.models
+WHERE name = 'salary_predictor';
+```
+
+Once the status is `complete`, the training phase is completed.
+
+### Validation Phase
+
+The `CREATE MODEL` statement performs validation of the model.
```
Comment (paxcema) [inline docs/model-types.mdx:102]: Maybe change to:

> By default, ...

File: docs/model-types.mdx [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+| 2020-07-02  | 443292.5194585953 | {"predicted_value": 443292.5194585953, "confidence": 0.9991, "anomaly": null, "truth": null, "confidence_lower_bound": 424501.59192981094, "confidence_upper_bound": 462083.4469873797}       |
+| 2020-10-02  | 443292.5194585953 | {"predicted_value": 443292.5194585953, "confidence": 0.9991, "anomaly": null, "truth": null, "confidence_lower_bound": 424501.59192981094, "confidence_upper_bound": 462083.4469873797}       |
++-------------+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+For details, check out [this tutorial](/sql/tutorials/house-sales-forecasting/).
+
+## How it Works in the Background
+
+MindsDB uses the Lightwood ML engine by default. The so-called mixers are used to create models by detecting patterns and behaviors of data. Here are [the mixers used by Lightwood](https://lightwood.io/mixer.html).
```
Comment (paxcema) [inline docs/model-types.mdx:384]: Let's reinforce the fact this is Lightwood-exclusive and may be different in other engines:

> MindsDB uses the Lightwood ML engine by default. This section takes a closer look at how this package automatically chooses what type of model to use. 

> Models in Lightwood follow an _encoder-mixer-decoder_ pattern, where refined (i.e. encoded) representations of all features are _mixed_ to produce target predictions. Here are...

Comment (paxcema) [reply docs/model-types.mdx:384]: Also, it's probably best to mention there is an ensembling step after training all mixers.

File: docs/model-types.mdx [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+A good example is the [Home Rentals model](/sql/tutorials/home-rentals#training-a-predictor/).
+
+</Tab>
+
+</Tabs>
+
+<br></br>
+
+<Note>
+MindsDB may use one or multiple mixers while preparing a model. Depending on the model type and the data type of the target variable, one mixer or a set of mixers is chosen to create, train, and validate an ML model.
```
Comment (paxcema) [inline docs/model-types.mdx:445]: We could mention ensembling here, fits well:

> [...] one mixer is chosen or a set of mixers are ensembled to create [...]

File: docs/model-types.mdx [hunk 4]
```diff
@@ -0,0 +1,467 @@
+---
+title: ML Models in MindsDB
+---
+
+In this section, we go through the ML model types available in MindsDB. These are regression models, classification models, and time series models.
+
+## What is an ML Model?
+
+A machine learning (ML) model is a program trained using the available data in order to learn how to recognize patterns and behaviors and make decisions that lead to predicting future data.
```
Comment (paxcema) [inline docs/model-types.mdx:9]: behaviors ~and make decisions that lead~ to predict future data.

Comment (paxcema) [inline docs/model-types.mdx:9]: I would rephrase the start as:

> A supervised machine learning (ML) [...]

File: docs/model-types.mdx [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+
+You input data into an ML model. It processes this input data, searching for patterns and correlations. After that, the ML model returns output data defined based on the input data.
+
+### Features
+
+Features define the columns or variables that an ML model uses as input data to search for patterns and predict the target variable.
```
Comment (paxcema) [inline docs/model-types.mdx:15]: ~define the columns or~ are

Comment (paxcema) [reply docs/model-types.mdx:15]: Add at the end:

> In tabular datasets, features usually correspond to single columns.

File: docs/model-types.mdx [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+
+### Target
+
+A target is a column or variable that an ML model predicts based on the information fetched out of the features.
```
Comment (paxcema) [inline docs/model-types.mdx:19]: ~A~ The target is a ~column or~ variable of interest that [...]

File: docs/model-types.mdx [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+
+### Training Dataset
+
+The training dataset is used during the training phase of an ML model. It contains both feature variables and a target variable. As its name indicates, it is used to train an ML model.
+
+The ML model takes the entire training dataset as input. It learns the patterns and relationships between feature variables and correlates them with the target variable values.
```
Comment (paxcema) [inline docs/model-types.mdx:25]: and ~correlates them with the~ target ~variable~ values.

File: docs/model-types.mdx [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+
+Once the training process is complete, one can move on to the validation phase.
+
+### Validation Dataset
+
+The validation dataset is used during the validation phase of an ML model. It contains both feature variables and a target variable, like the training dataset. But as its name indicates, it is used to validate the predictions made by an ML model.
```
Comment (paxcema) [inline docs/model-types.mdx:31]: Add: 

> It has no overlap with the training dataset, as it is a "held-out" set to simulate a real scenario where the model generates predictions for novel input data.

File: docs/model-types.mdx [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+
+The ML model takes only the feature variables from the validation dataset as input. Based on what the model learns during the training process, it makes predictions for the values of a target variable.
+
+Now comes the validation step. To assess the accuracy of the ML model, one compares the target variable values from the validation dataset with the target variable values predicted by the ML model. The closer these values are to each other, the better accuracy of the ML model.
+
+### Input Dataset
+
+After completing the training and validation phases, one can provide the input dataset consisting of only the feature variables to predict the target variable values.
+
+## How an ML Model is Created?
```
Comment (paxcema) [inline docs/model-types.mdx:41]: Change to:

> ## How is an ML model created?

File: docs/model-types.mdx [hunk 10]
```diff
@@ -0,0 +1,467 @@
...
+|salary|salary_explain                                                                                                                            |
++------+------------------------------------------------------------------------------------------------------------------------------------------+
+|128   |{"predicted_value": 128, "confidence": 0.67, "anomaly": null, "truth": null, "confidence_lower_bound": 109, "confidence_upper_bound": 147}|
++------+------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+By comparing the real salary values for the defined individuals and the predicted salary values, one can figure out the accuracy of the ML model.
+
+<Note>
+Please note that MindsDB calculates the model's accuracy while running the `CREATE MODEL` statement.
```
Comment (paxcema) [inline docs/model-types.mdx:131]: > Please note that by default [...]

Because it's not guaranteed that all ML engines will do this.

File: docs/model-types.mdx [hunk 11]
```diff
@@ -0,0 +1,467 @@
...
+
+The `CREATE MODEL` statement does the following:
```
Comment (paxcema) [inline docs/model-types.mdx:133]: Same as above ^

File: docs/model-types.mdx [hunk 12]
```diff
@@ -0,0 +1,467 @@
...
++-------+---------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Yes   | 0.7752808988764045  | {"predicted_value": "Yes", "confidence": 0.7752808988764045, "anomaly": null, "truth": null, "probability_class_No": 0.4756, "probability_class_Yes": 0.5244}    |
++-------+---------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+For details, check out [this tutorial](/sql/tutorials/customer-churn/).
+
+### Time Series Models
+
+Time series models fall under the regression or classification category. But what's distinct about them is that we order data by date, time, or any value defining sequential order of events.
```
Comment (paxcema) [inline docs/model-types.mdx:295]: Maybe add:
> Usually, predictions made by time series models are referred to as "forecasts".

File: docs/model-types.mdx [hunk 13]
```diff
@@ -0,0 +1,467 @@
...
+
+<br></br>
+
+<Note>
+MindsDB may use one or multiple mixers while preparing a model. Depending on the model type and the data type of the target variable, one mixer or a set of mixers is chosen to create, train, and validate an ML model.
+</Note>
+
+The three cases above describe how MindsDB chooses the mixer candidates and stores them in the `submodels` array.
+
+By default, after collecting all relevant mixers in the `submodels` array, MindsDB uses the [BestOf](https://lightwood.io/ensemble.html#ensemble.BestOf) ensemble to [choose one of these mixers to fit the model](https://github.com/mindsdb/lightwood/blob/9a48d5523d78111c77024e1983d281f37a9782ea/lightwood/api/json_ai.py#L353,L358).
```
Comment (paxcema) [inline docs/model-types.mdx:450]: after ~collecting~ training all [...]

Comment (paxcema) [inline docs/model-types.mdx:450]: ensemble to [single out the best mixer as the final model]

File: docs/model-types.mdx [hunk 14]
```diff
@@ -0,0 +1,467 @@
...
+
+But you can always use a different ensemble that allows using multiple mixers per model, such as the [MeanEnsemble](https://lightwood.io/ensemble.html#ensemble.MeanEnsemble), [ModeEnsemble](https://lightwood.io/ensemble.html#ensemble.ModeEnsemble), [StackedEnsemble](https://lightwood.io/ensemble.html#ensemble.StackedEnsemble), [TsStackedEnsemble](https://lightwood.io/ensemble.html#ensemble.TsStackedEnsemble), or [WeightedMeanEnsemble](https://lightwood.io/ensemble.html#ensemble.WeightedMeanEnsemble) ensemble type.
```
Comment (paxcema) [inline docs/model-types.mdx:452]: that ~allows using~ may aggregate multiple mixers per model

**Review by paxcema — CHANGES_REQUESTED** [review]
Left some corrections and suggestions. For the next PR similar to this one (lots of new text) I'd recommend that we collaborate on a shared doc instead, so that it's easier to suggest/edit directly 🤓 

A general caveat I have is that we should probably communicate at the very start that a lot of these behaviors are handler-specific, and that we're describing the default behavior (which boils down to what is Lightwood and how can it be used). It's possible that other handlers will behave differently (for example, some of them may not perform validation automatically when creating a model).

Also, it's worth highlighting that the deep dive at the end is directly referencing lightwood code that may change substantially in the coming months, just to be aware of this and keep track 👀.

File: docs/model-types.mdx (hunk 1 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:102]: Changed to _By default, the `CREATE MODEL` statement performs validation of the model._

File: docs/model-types.mdx (hunk 2 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:384]: Modified it like this:

> ## How it Works in the Background
> 
> MindsDB uses the Lightwood ML engine by default. This section takes a closer look at how this package automatically chooses what type of model to use.
> 
> Models in Lightwood follow an encoder-mixer-decoder pattern, where refined, or encoded, representations of all features are mixed to produce target predictions. Here are [the mixers used by Lightwood](https://lightwood.io/mixer.html).
> 
> Please note that there is an ensembling step after training all mixers. Read on to learn more.
> 
> To give you some details on how MindsDB creates a model using different mixers, here is [the full code](https://github.com/mindsdb/lightwood/blob/9a48d5523d78111c77024e1983d281f37a9782ea/lightwood/api/json_ai.py).

File: docs/model-types.mdx (hunk 3 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:445]: Changed:

> <Note>
> MindsDB may use one or multiple mixers while preparing a model. Depending on the model type and the data type of the target variable, one mixer is chosen or a set of mixers are ensembled to create, train, and validate an ML model.
> </Note>

File: docs/model-types.mdx (hunk 4 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:9]: Changed:

> A machine learning (ML) model is a program trained using the available data in order to learn how to recognize patterns and behaviors to predict future data.

Comment (martyna-mindsdb) [reply docs/model-types.mdx:9]: Changed:

> ## What is an ML Model?
> 
> A machine learning (ML) model is a program trained using the available data in order to learn how to recognize patterns and behaviors to predict future data. There are various types of ML models that use different learning paradigms, but MindsDB models are all *supervised* because they learn from pairs of input data and expected output.
> 
> You input data into an ML model. It processes this input data, searching for patterns and correlations. After that, the ML model returns output data defined based on the input data.

File: docs/model-types.mdx (hunk 5 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:15]: Changed:

> ### Features
> 
> Features are variables that an ML model uses as input data to search for patterns and predict the target variable. In tabular datasets, features usually correspond to single columns.

File: docs/model-types.mdx (hunk 6 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:19]: Changed:

> ### Target
> 
> The target is a variable of interest that an ML model predicts based on the information fetched out of the features.

File: docs/model-types.mdx (hunk 7 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:25]: Changed:

> The ML model takes the entire training dataset as input. It learns the patterns and relationships between feature variables and target values.

File: docs/model-types.mdx (hunk 9 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:41]: Changed (but kept the capitalized letters to be consistent):

> ## How is an ML Model Created?

File: docs/model-types.mdx (hunk 10 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:131]: Changed:

> Please note that MindsDB calculates the model's accuracy by default while running the `CREATE MODEL` statement. However, it is not guaranteed that all ML engines do this.

File: docs/model-types.mdx (hunk 11 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:133]: Changed:

> By default, the `CREATE MODEL` statement does the following:

File: docs/model-types.mdx (hunk 12 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:295]: Changed:

> ### Time Series Models
> 
> Time series models fall under the regression or classification category. But what's distinct about them is that we order data by date, time, or any value defining sequential order of events. Usually, predictions made by time series models are referred to as *forecasts*.

File: docs/model-types.mdx (hunk 13 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:450]: Changed:

> By default, after training all relevant mixers in the `submodels` array, MindsDB uses the [BestOf](https://lightwood.io/ensemble.html#ensemble.BestOf) ensemble to [choose one of these mixers to fit the model](https://github.com/mindsdb/lightwood/blob/9a48d5523d78111c77024e1983d281f37a9782ea/lightwood/api/json_ai.py#L353,L358).

Comment (martyna-mindsdb) [reply docs/model-types.mdx:450]: Changed:

> By default, after training all relevant mixers in the `submodels` array, MindsDB uses the [BestOf](https://lightwood.io/ensemble.html#ensemble.BestOf) ensemble to [single out the best mixer as the final model](https://github.com/mindsdb/lightwood/blob/9a48d5523d78111c77024e1983d281f37a9782ea/lightwood/api/json_ai.py#L353,L358).

File: docs/model-types.mdx (hunk 14 above)
Comment (martyna-mindsdb) [reply docs/model-types.mdx:452]: Changed:

> But you can always use a different ensemble that may aggregate multiple mixers per model, such as the [MeanEnsemble](https://lightwood.io/ensemble.html#ensemble.MeanEnsemble), [ModeEnsemble](https://lightwood.io/ensemble.html#ensemble.ModeEnsemble), [StackedEnsemble](https://lightwood.io/ensemble.html#ensemble.StackedEnsemble), [TsStackedEnsemble](https://lightwood.io/ensemble.html#ensemble.TsStackedEnsemble), or [WeightedMeanEnsemble](https://lightwood.io/ensemble.html#ensemble.WeightedMeanEnsemble) ensemble type.

**Comment by martyna-mindsdb** [comment]
@paxcema 
I agree. Next time, I'll create a Google doc to ease the process.

I added this at the beginning:
image

I'll keep it on my task list to check for any Lightwood code changes in the future and keep this doc page up to date.

**Review by paxcema — APPROVED** [review]

**Review by ZoranPandovski — APPROVED** [review]
